File: examples/python/07-poo/06-static_method.rb
#!/usr/bin/env ruby

class Employee
  def self.sample(x)
    puts "Inside static method #{x}"
  end
end

# call static method
Employee.sample 10

# can't  be called using object
emp = Employee.new
begin
  emp.sample 10
rescue NoMethodError
  puts "NoMethodError: static methods belongs to Class not to instances"
end

File: examples/python/17-poo/06-static_method.py
#!/usr/bin/env python3
# https://pynative.com/python-static-method/

class Employee:
    @staticmethod
    def sample(x):
        print('Inside static method', x)

# call static method
Employee.sample(10)

# can be called using object
emp = Employee()
emp.sample(10)

